File: MojStomatolog/MojStomatolog.Models/Core/Appointment.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace MojStomatolog.Models.Core
{
    public class Appointment
    {
        public int AppointmentId { get; set; }
        public DateTime AppointmentDateTime { get; set; }
        public string Procedure { get; set; } = null!;
        public bool IsConfirmed { get; set; }
        public string Notes { get; set; } = null!;
        public int PatientId { get; set; }
        public int EmployeeId { get; set; }

        [ForeignKey("PatientId")]
        public User? Patient { get; set; }

        [ForeignKey("EmployeeId")]
        public Employee? Employee { get; set; }
    }
}

File: MojStomatolog/UI/mojstomatolog_desktop/lib/models/appointment.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:mojstomatolog_desktop/models/employee.dart';
import 'package:mojstomatolog_desktop/models/service.dart';
import 'package:mojstomatolog_desktop/models/user.dart';

part 'appointment.g.dart';

@JsonSerializable()
class Appointment {
  int? appointmentId;
  DateTime? appointmentDateTime;
  int? serviceId;
  bool? isConfirmed;
  String? notes;
  int? patientId;
  int? employeeId;

  Service? service;
  User? patient;
  Employee? employee;

  Appointment();

  factory Appointment.fromJson(Map<String, dynamic> json) =>
      _$AppointmentFromJson(json);
  Map<String, dynamic> toJson() => _$AppointmentToJson(this);
}
